### PR TITLE
Refactor HTTPoison wrapper

### PR DESCRIPTION
**Pull request by pinx**
This is just to show what the code could look like. Some more changes need to be made.
We need more tests.
Because you "use HTTPoison.Base", you don't have to override all the functions. Also, config can be simpler.

### CODE REVIEW

**Comment by austinsmorris** [comment]
Hello, @pinx! This is your first Pull Request that will be reviewed by [Ebert](https://ebertapp.io/), an automatic Code Review service. It will leave comments on this diff with potential issues and style violations found in the code as you push new commits. You can also see all the issues found on this Pull Request on its [review page](https://ebertapp.io/github/austinsmorris/arangoex/pulls/4). Please check our [documentation](https://ebertapp.io/docs) for more information.

File: lib/arangoex.ex [hunk 1]
```diff
@@ -1,58 +1,33 @@
...
-    [@base_url, url]
+  def process_url(url) do
+    @base_url <> url
   end
 
-  def delete(url, headers \\ [], options \\ []) do
-    response = request(:delete, url, "", get_headers(headers), options)
-    response |> Response.convert_response()
+  def process_response_body(body) do
+    #TODO: maybe do some processing?
```
Comment (austinsmorris) [inline lib/arangoex.ex]: TODO found

**Comment by austinsmorris** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 1 possible new issue (including those that may have been commented here).

You can see more details about this review at https://ebertapp.io/github/austinsmorris/arangoex/pulls/4.

**Comment by pinx** [comment]
I would move the HTTPoison wrapper to a module Arangoex.Db, and follow José's advice on mocking: http://blog.plataformatec.com.br/2015/10/mocks-and-explicit-contracts/
The MockDb would return predefined jsons for documents, graphs, etc.

**Pull request closed without merging**